### PR TITLE
Move Orca's GUCs into the extension

### PR DESCRIPTION
**Pull request by whitehawk**
Move Orca's GUCs into the extension

This patch:
1. Moves all GUCs, that are related to Orca and used inside it, into Orca
extension.
2. Updates minirepro utility tool. This tool generates a detailed information
related to the monitored SQL commands being executed, including non-default
GUCs. The patch updates the query used to retrieve non-default GUCs. The update
is needed, as, after moving the "optimizer_*" GUCs, their 'category' in
'pg_settings' table had been changed to 'Customized Options' value. Without this
change the respective regression test 'minirepro' fails.
3. Adds all the touched GUC names into the file with ignored symbols for ABI
check, as they no longer exist in the main executable.
4. Removes GUC 'optimizer_partition_selection_log' completely, as it is not used
anywhere.

Note. Following GUCs were not moved, as they are used inside the core:
 - 'optimizer_enable_query_parameter'
 - 'optimizer_analyze_root_partition'
 - 'optimizer_analyze_midlevel_partition'
 - 'optimizer_replicated_table_insert'
 - 'optimizer'
 - 'optimizer_control'

### CODE REVIEW

File: gpcontrib/orca/orca_guc.c
Comment (dkovalev1) [inline gpcontrib/orca/orca_guc.c:1]: Formatting of this file has changed and does not correspond neither source guc_gp.c nor pgintent.

**Comment by dkovalev1** [comment]
perhaps parameters 
```
bool		optimizer;
bool		optimizer_control = true;
```
also left in the core and have to be added to the list in the description

**Comment by whitehawk** [comment]
> perhaps parameters
> 
> ```
> bool		optimizer;
> bool		optimizer_control = true;
> ```
> 
> also left in the core and have to be added to the list in the description

Yes, added them as well to the description. Missed them previously because there is already a separate ticket for the `optimizer` GUC.

File: gpcontrib/orca/orca_guc.c
Comment (whitehawk) [reply gpcontrib/orca/orca_guc.c:1]: On one of preceding PRs we decided to use clang format for 'c' sources in orca extension to be consistent (as orca's c++ code already uses it). So it is clang-formatted.

**Review by dkovalev1 — APPROVED** [review]

**Review by silent-observer — APPROVED** [review]
LGTM

**Pull request merged**